classes/security/Validation.php

```diff
@@ -450,73 +453,76 @@ public static function canAdminister($administeredUserId, $administratorUserId)
+            ->getCount() > 0;
+
+        if ($isAdministratorSiteAdmin) {
             return self::ADMINISTRATION_FULL;
         }
 
-        // Make sure the administering user has a manager role somewhere
-        $roleManagerCount = Repo::userGroup()
+        // ensure administrator have a manager role somewhere
+        $hasManagerRole = Repo::userGroup()
             ->getCollector()
             ->filterByUserIds([$administratorUserId])
             ->filterByRoleIds([Role::ROLE_ID_MANAGER])
-            ->getCount();
+            ->getCount() > 0;
 
-        if ($roleManagerCount <= 0) {
+        if (!$hasManagerRole) {
             return self::ADMINISTRATION_PROHIBITED;
         }
 
-        $administeredUserAssignedGroupIds = Repo::userGroup()
-            ->getCollector()
-            ->filterByUserIds([$administeredUserId])
-            ->getMany()
-            ->map(fn ($userGroup) => $userGroup->getContextId())
-            ->sort()
-            ->toArray();
-
-        $administratorUserAssignedGroupIds = Repo::userGroup()
-            ->getCollector()
-            ->filterByUserIds([$administratorUserId])
-            ->filterByRoleIds([Role::ROLE_ID_MANAGER])
-            ->getMany()
-            ->map(fn ($userGroup) => $userGroup->getContextId())
-            ->sort()
-            ->toArray();
-
-        // Check for administered user group assignments in other contexts
-        // that the administrator user doesn't have a manager role in.
-        if (collect($administeredUserAssignedGroupIds)->diff($administratorUserAssignedGroupIds)->count() > 0) {
-            // Found an assignment: disqualified.
-            // But also determine if a partial administrate is allowed
-            // if the Administrator User is a Journal Manager in the current context
-            if ($contextId !== null &&
+        // optimize query to check for unmanaged contexts
+        $unmanagedContextsExist = \Illuminate\Support\Facades\DB::table('user_user_groups as uug_administered')
+            ->join('user_groups as ug_administered', 'uug_administered.user_group_id', '=', 'ug_administered.user_group_id')
+            ->where('uug_administered.user_id', $administeredUserId)
+            ->whereNotIn('ug_administered.context_id', function ($query) use ($administratorUserId) {
+                $query->select('ug_administrator.context_id')
+                    ->from('user_user_groups as uug_administrator')
+                    ->join('user_groups as ug_administrator', 'uug_administrator.user_group_id', '=', 'ug_administrator.user_group_id')
+                    ->where('uug_administrator.user_id', $administratorUserId)
+                    ->where('ug_administrator.role_id', Role::ROLE_ID_MANAGER);
+            })
+            ->exists();
+
+        if ($unmanagedContextsExist) {
+            // check if partial administration is allowed in the current context
+            $isManagerInCurrentContext = $contextId !== null &&
```

Comment:
Have you tested this? in my case `$contextId` is null here. Steps to debug: 
1. Use test dataset
2. Create a second journal
3. Assign a user with a sub editor role another role in the second journal
4. Go to settings -> users in the first journal by a user that has only manager role in that journal.

Reply:
Thanks, @touhidurabir. What is current business logic behind `login as` and partial administration? Who can administer whom and what additional actions could/couldn't be done, e.g., changing the role, merging  users... Is there a good description somewhere?